### PR TITLE
feat: added framework for remote chan tests and TC for token transfer …

### PR DESCRIPTION
**Pull request by bcsainju**
### Commit Message

```bash
feat(test): added framework for remote chan tests and TCs for token transfer to-and-from ibc-centauri
```

see the [guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages) for commit messages.

### Changelog Entry

```bash
version: <log entry>
```

## Checklist

- [x] I have performed a self-review of my own code
- [ ] I have documented my code in accordance with the [documentation guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#documentation)
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the unit tests
- [ ] I only have one commit (if not, squash them into one commit).
- [ ] I have a descriptive commit message that adheres to the [commit message guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages)
- [ ] I have added version bump label on PR.

> Please review the [CONTRIBUTING.md](/CONTRIBUTING.md) file for detailed contributing guidelines.

### CODE REVIEW

File: test/interchain.go [hunk 1]
```diff
@@ -156,33 +155,32 @@ type InterchainLink struct {
...
-		panic(fmt.Errorf("chain with name=%s and id=%s was never added to Interchain", cfg.Name, cfg.ChainID))
-	}
+	// if _, exists := ic.chains[link.Chains[0].(ibc.Chain)]; !exists {
+	// 	cfg := link.Chains[0].(ibc.Chain).Config()
+	// 	panic(fmt.Errorf("chain with name=%s and id=%s was never added to Interchain", cfg.Name, cfg.ChainID))
+	// }
+	// if _, exists := ic.chains[link.Chains[1].(ibc.Chain)]; !exists {
+	// 	cfg := link.Chains[1].(ibc.Chain).Config()
+	// 	panic(fmt.Errorf("chain with name=%s and id=%s was never added to Interchain", cfg.Name, cfg.ChainID))
+	// }
```
Comment (ibrizsabin) [inline test/interchain.go:165]: lets remove these comments if not necessary.

File: test/interchain.go [hunk 2]
```diff
@@ -156,33 +155,32 @@ type InterchainLink struct {
...
 	if _, exists := ic.relayers[link.Relayer]; !exists {
 		panic(fmt.Errorf("relayer %v was never added to Interchain", link.Relayer))
 	}
 
-	if link.Chain1 == link.Chain2 {
-		panic(fmt.Errorf("chains must be different (both were %v)", link.Chain1))
-	}
+	// if link.Chains[0] == link.Chains[1] {
+	// 	panic(fmt.Errorf("chains must be different (both were %v)", link.Chains[0]))
+	// }
```
Comment (ibrizsabin) [inline test/interchain.go:173]: same here

File: test/chains/icon/localnet.go
[the remark this answers is not in the document]
Comment (bcsainju) [reply test/chains/icon/localnet.go:1]: some methods are required just to comply with the interchain test framework chain interface which uses docker internally but we don;t need all the methods, so some of them are just there to comply with the interface declaration.

File: test/interchain.go (hunk 1 above)
Comment (bcsainju) [reply test/interchain.go]: done.

**Review by ibrizsabin — APPROVED** [review]

**Pull request merged**